opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -37,6 +38,12 @@
 
 namespace Opm {
 
+    /// Returns a copy of s where every letter has been upper cased.
+    static std::string toupper(const std::string& s) {
+        auto ret = std::string(s);
+        std::transform(ret.begin(), ret.end(), ret.begin(), ::toupper);
```

Comment:
It's one of those cases™ where it depends on whether or not it's provided by types.h, ctypes or something. You can always force it to be a function pointer context.

Regardless, I had a chat with Pål and we'll soon create a separate PR that provides some of the string functions that were thrown out earlier.

Some context: In a refactoring a lot of boost-string use was replaced by a simple inlined transform, because we really only needed to uppercase things once, and the refactoring made data flow easier to follow. Now, on the other hand, the strings being uppercase'd come through a completely different access path, so we cannot rely on input reading doing it for us.